packages/order-utils/src/order_validation_utils.ts

```diff
@@ -210,13 +219,30 @@ export class OrderValidationUtils {
         const desiredFillTakerTokenAmount = remainingTakerTokenAmount.lessThan(fillTakerAssetAmount)
             ? remainingTakerTokenAmount
             : fillTakerAssetAmount;
-        await OrderValidationUtils.validateFillOrderBalancesAllowancesThrowIfInvalidAsync(
-            exchangeTradeEmulator,
-            signedOrder,
-            desiredFillTakerTokenAmount,
-            takerAddress,
-            zrxAssetData,
-        );
+        try {
+            await OrderValidationUtils.validateFillOrderBalancesAllowancesThrowIfInvalidAsync(
+                exchangeTradeEmulator,
+                signedOrder,
+                desiredFillTakerTokenAmount,
+                takerAddress,
+                zrxAssetData,
+            );
+        } catch (err) {
+            const transferFailedErrorMessages = [
+                ExchangeContractErrs.InsufficientMakerBalance,
+                ExchangeContractErrs.InsufficientMakerFeeBalance,
+                ExchangeContractErrs.InsufficientTakerBalance,
+                ExchangeContractErrs.InsufficientTakerFeeBalance,
+                ExchangeContractErrs.InsufficientMakerAllowance,
+                ExchangeContractErrs.InsufficientMakerFeeAllowance,
```

Comment:
As part of the larger refactor we should also remember to remove the older `ExchangeContractErrs` which are no longer applicable.

Reply:
Ditto above.